<?xml version="1.0" ?>
<launch>
  
    <arg name="ns" default="iris"/>

    <!--
    <include file="$(find rotors_gazebo)/launch/gazebo_iris_aircraft_world.launch">
	    <arg name="ns" value="$(arg ns)"/>
    </include>
    -->
    
    <arg name="headless" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="enable_logging" default="false"/>
    <arg name="enable_ground_truth" default="true"/>
    <arg name="log_file" default="$(arg ns)"/>

    <arg name="mav_name" default="iris"/>

    
    
    <env name="GAZEBO_MODEL_PATH" value="$(find rotors_gazebo)/models"/>
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
	<arg name="world_name" value="$(find rotors_gazebo)/worlds/etihad_aircraft_world_half.world"/> <!-- half sized -->
	<arg name="debug" value="false" />
	<arg name="headless" value="$(arg headless)"/>
	<arg name="gui" value="$(arg gui)"/>
    </include>
    

    <group ns="$(arg mav_name)">
	<include file="$(find rotors_gazebo)/launch/spawn_iris.launch">
	    <arg name="model" value="$(find rotors_description)/urdf/$(arg mav_name)_surface_sensors.gazebo" />
	    <arg name="enable_logging" value="$(arg enable_logging)" />
	    <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
	    <arg name="log_file" value="$(arg log_file)"/>
	    <arg name="Y" value="3.14"/>
	    <arg name="y" value="-15"/><!-- "-30" -->
	    <arg name="x" value="4"/><!-- "4" -->
	</include>
    </group>
    
    
    
    
    
    <include file="$(find px4)/launch/iris.launch">
	<arg name="ns" value="$(arg ns)"/>
    </include>


    <include file="$(find px4)/launch/mavros_sitl.launch">
	    <arg name="ns" value="$(arg ns)"/>
    </include>



    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0.0 -0.5 0.5 -0.5 0.5 /iris/xtion_sensor/ground_truth/iris/xtion_sensor/ground_truth/odometry_sensor_link /iris/xtion_sensor/camera_depth_optical_frame 500" />
    <!-- <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0.0 -0.5 0.5 -0.5 0.5 /iris/camera_sensor/ground_truth/iris/camera_sensor/ground_truth/odometry_sensor_link /iris/camera_sensor/camera_link 500" />-->

    <node pkg="rviz" name="rviz" type="rviz" args="-d $(find aircraft_inspection)/rviz/nbv.rviz" required="true" />
    <!--<node pkg="aircraft_inspection" name="nbv_loop" type="nbv_loop" args="" required="true" />-->
</launch>
